doc/community/developer-guide.md

```diff
@@ -233,65 +202,22 @@ To test the output:
 assert out.result == 2
 ```
 
-You can also check for execution success:
-
-```{code-cell} ipython3
-assert out.success
-```
-
-```{important}
-Always check for success! Since `run_cell` won't raise an error if the code fails
-```
-
-```{code-cell} ipython3
-try:
-    ip_session.run_cell("1 / 0")
-except Exception as e:
-    print(f"Error: {e}")
-else:
-    print("No error")
-```
-
-Note that the `run_cell` only printed the error but did not raise an exception.
-
-+++
-
-#### Capturing errors
-
-Let's see how to test that the code raises an expected error:
-
-```{code-cell} ipython3
-out = ip_session.run_cell("1 / 0")
-```
-
-```{code-cell} ipython3
-# this returns the raised exception
-out.error_in_exec
-```
-
-```{code-cell} ipython3
-:tags: [raises-exception]
-
-# this raises the error
-out.raise_error()
-```
-
-You can then use pytest to check the error:
+You can then use pytest to check for errors:
 
 ```{code-cell} ipython3
 import pytest
 ```
 
 ```{code-cell} ipython3
 with pytest.raises(ZeroDivisionError):
-    out.raise_error()
+    ip_session.run_cell("1 / 0")
 ```
 
 To check the error message:
 
 ```{code-cell} ipython3
 with pytest.raises(ZeroDivisionError) as excinfo:
-    out.raise_error()
+    ip_session.run_cell("1 / 0")
```

Comment:
Would it help to add an example of creating custom `ip` objects that initialize specific data (and also add reference to some existing fixtures that create custom `ip`)

Reply:
it's rarely the case that people have to create new `ip`. there is already a reference to available `ip`s in the developer docs.